### ISSUE TITLE
Counting from HDF5 file causes kernel to die

### ISSUE THREAD

**Issue by ercius** [OpenChemistry/stempy#219]
Counting from an HDF5 file does not work. The python kernel just dies.  This is not a common operation, but it should not just fail.

Example code:
```python
import stempy.io as stio
import stempy.image as stim
import numpy as np

dark0 = np.zeros((576,576))

with h5py.File(fPath, 'r') as f0:
    sReader = stio.reader(f0)
    ee = stim.electron_count(sReader,dark0,number_of_samples=1200,
                                            verbose=True,
                                            xray_threshold_n_sigma=175,
                                            background_threshold_n_sigma=threshold)
```

**Comment by psavery** [comment]
@ercius I gave this script a try just now on an HDF5 dataset that I have here locally (that contains raw data), and it ran without an issue. Can you provide me with the dataset you are using?

**Comment by ercius** [comment]
Odd. Ill double check I can reproduce. Ill share the dataset if it still happens. Ill get back to you soon.

**Comment by ercius** [comment]
Ok. I double checked by creating a new HDF5 file from the test scan #212 currently available on the system. The counting worked so my issue must have been something else.

However, if you use `stempy.io.write_hdf5` to create an HDF5 file then you get an error that the `stem/images` HDF5 dataset does not exist. A fix would be to finish #133 , make sure that `stempy.io.write_hdf5` writes the `scan_dimensions` attribute properly and `stempy.image.electron_count` uses that attribute instead of the shape of the `stem/images` dataset.
